src/Workspaces/Core/Portable/Rename/Renamer.cs

```diff
@@ -25,6 +26,96 @@ public static class Renamer
                 newName, optionSet, cancellationToken);
         }
 
+        /// <summary>
+        /// Similar to calling <see cref="Document.WithName(string)" /> with additional changes to the solution. 
+        /// Each change is added as a <see cref="RenameDocumentAction"/> in the returned <see cref="RenameDocumentActionSet.ApplicableActions" />.
+        /// 
+        /// Each action may individually encounter errors that prevent it from behaving correctly. Those are reported in <see cref="RenameDocumentAction.GetErrors(System.Globalization.CultureInfo?)"/>.
+        /// 
+        /// Current supported actions that may be returned: 
+        /// * <see cref="RenameSymbolDocumentAction"/> that will rename the type to match the document name
```

Comment:
we don't offer to sync namespaces if you rename the document?

Reply:
No, they are separate APIs. Namespace is based on document folders

Reply:
ok.  i don't see why we need the action sets here.

Reply:
I believe the intent is more is coming too -- you might do both renaming a symbol _and_ also updating file copyright headers, or want to update just the copyright header but the type is public so you can't change that, etc.

Reply:
RenameSymbolDocumentAction is an internal type.  So this `see` won't go anywhere for people looking at docs.